drivers/android/process.rs

```diff
@@ -186,10 +186,8 @@ impl ProcessInner {
         strong: bool,
         thread: Option<&Thread>,
     ) -> KernelResult<Option<NodeRef>> {
-        Ok(match self.get_existing_node(ptr, cookie)? {
-            None => None,
-            Some(node) => Some(self.new_node_ref(node, strong, thread)),
-        })
+        self.get_existing_node(ptr, cookie)?
```

Comment:
Don't you need to wrap this in `Ok()`?

Reply:
I'm actually confused about the build process. I used `make -j 4 CLIPPY=1`, and put non compiling rust code (usually wouldn't build), and the build process said the kernel was still ready...

Reply:
J4 dıl benım cıhazım redmı 6

Reply:
> I'm actually confused about the build process. I used `make -j 4 CLIPPY=1`, and put non compiling rust code (usually wouldn't build), and the build process said the kernel was still ready...

Ideally you'd try patches out before sending pull requests.

In any case, read the 'Configuration' part of https://github.com/Rust-for-Linux/linux/blob/rust/Documentation/rust/quick-start.rst -- you need both rust and binder enabled to build this code.

Reply:
Perfect, new to kernel hacking. Fixed this issue/building/tested in qemu (although I don't know if testing in x86 will touch this code)

Reply:
Thanks!

I think it may be better if you squash the two commits into one given that the first one doesn't build on its own and may be problematic if we're bisecting.

Reply:
+1 Please squash them @wcampbell0x2a.

Reply:
> +1 Please squash them @wcampbell0x2a.

done